ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageServiceFactory.java

```diff
@@ -85,11 +83,15 @@ public BlobStorageService getBlobStorageService() {
       AccountService accountService = accountServiceFactory.getAccountService();
       IdConverterFactory idConverterFactory =
           Utils.getObj(frontendConfig.frontendIdConverterFactory, verifiableProperties, clusterMap.getMetricRegistry());
+      UrlSigningServiceFactory urlSigningServiceFactory =
+          Utils.getObj(frontendConfig.frontendUrlSigningServiceFactory, verifiableProperties,
+              clusterMap.getMetricRegistry());
+      UrlSigningService urlSigningService = urlSigningServiceFactory.getUrlSigningService();
       SecurityServiceFactory securityServiceFactory =
           Utils.getObj(frontendConfig.frontendSecurityServiceFactory, verifiableProperties,
-              clusterMap.getMetricRegistry(), accountService);
+              clusterMap.getMetricRegistry(), accountService, urlSigningService);
       return new AmbryBlobStorageService(frontendConfig, frontendMetrics, responseHandler, router, clusterMap,
-          idConverterFactory, securityServiceFactory, accountService);
+          idConverterFactory, securityServiceFactory, accountService, urlSigningService);
```

Comment:
minor: Some factories are sent as arg, where as for some, we fetch the instance and send in as arg. Should fix it in some later patch.